library/kotlin/src/io/envoyproxy/envoymobile/BUILD

```diff
@@ -60,3 +61,10 @@ envoy_mobile_kt_library(
         "//library/java/src/io/envoyproxy/envoymobile/engine:envoy_base_engine_lib",
     ],
 )
+
+detekt(
+    name = "envoy_lib_lint",
+    srcs = glob(["*.kt"]),
+    build_upon_default_config = True,
+    cfgs = ["//:kotlin_lint_config"],
+)
```

Comment:
Maybe we can just do a global `detekt` rule and just add it to the top level `BUILD` file. What do you think about that?

Reply:
Do you mean a single target with all Kotlin sources? Certainly will work. However, this gives Bazel a little room to run things in parallel. Thinking about — I probably should change the GitHub Actions workflow to query Detekt targets and run them in a single Bazel invocation.

Reply:
Gotcha. That sounds good. There might be a good way for us to hide this in a custom bazel macro in the future too.

Do you want to do the actions change in this pr as well or are you looking to do it as a follow up?

Reply:
Just tried to fight the CI. Seems like `bazel query` [evaluates a bit much and fails to resolve `swiftc` on a Linux machine](https://github.com/lyft/envoy-mobile/runs/613134683#step:4:50). Is it possible to resolve this somehow? I’ve looked over different GitHub workflows but saw no difference.

Reply:
I'll have to dig deeper to figure out (I end up needing to relearn bazel + ci every time). We can move forward without that niceness change for now.

Reply:
@artem-zinnatullin, do you have ideas how to resolve this? Or a better approach is just not to use `bazel query` for this at all?

Reply:
All right, removed `bazel query` for now, we can tune it in the future with the growing number of Kotlin targets.

Reply:
hm, this is not good indeed if `query` fails depending on OS, there is probably a way to use [config_settings](https://docs.bazel.build/versions/master/configurable-attributes.html) to only configure swift if we're running on macOS and then use `cquery` or something like that

Maybe @justhecuke, @keith or @kastiglione have some ideas

Reply:
What's the specific query failure? I expect it's because some of the repository rule setup in rules_swift checks what features your swift version supports, and those fail. I don't think we've had this issue internally since likely even if we weren't using it directly our docker images always had swift installed. It might be worth looking for a related bazel issue, seems like there must be one about this. Note that you also hit problems like this on macOS if you don't have the Android SDK installed